packages/sanity/src/core/form/inputs/files/ImageInput/ImageActionsMenu.tsx

```diff
@@ -35,20 +43,21 @@ export function ImageActionsMenu(props: ImageActionsMenuProps) {
         />
       )}
 
-      <MenuButton
-        button={
-          <Button
-            aria-label="Open image options menu"
-            data-testid="options-menu-button"
-            icon={EllipsisVerticalIcon}
-            mode="ghost"
-          />
-        }
+      <Popover
         id="image-actions-menu"
-        menu={<Menu>{children}</Menu>}
-        popover={POPOVER_PROPS}
-        ref={setMenuButtonElement}
-      />
+        content={<Menu ref={setMenuRef}>{children}</Menu>}
+        portal
+        open={isMenuOpen}
+        constrainSize
+      >
+        <Button
```

Comment:
We need to close the menu when pressing "Escape" and setting the focus back on the button. Something like:


```ts
  const [buttonElement, setButtonElement] = useState<HTMLButtonElement | null>(null)

  useGlobalKeyDown(
    useCallback(
      (event) => {
        if (isMenuOpen && event.key === 'Escape') {
          onMenuOpen(false)
        }
      },
      [isMenuOpen, onMenuOpen]
    )
  )
  
<Popover
  // rest of props...
>
  <Button
   // rest of props...
    ref={setButtonElement}
  />
</Popover>
  
  

```